src/corehost/cli/fxr_resolver.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef _COREHOST_CLI_FXR_RESOLVER_H_
+#define _COREHOST_CLI_FXR_RESOLVER_H_
+
+#include <pal.h>
+
+// Forward declaration of required custom feature APIs
+using hostfxr_main_fn = int(*)(const int argc, const pal::char_t* argv[]);
+using hostfxr_main_startupinfo_fn = int(*)(
+    const int argc,
+    const pal::char_t* argv[],
+    const pal::char_t* host_path,
+    const pal::char_t* dotnet_root,
+    const pal::char_t* app_path);
+using hostfxr_get_delegate_fn = int(*)(
+    const pal::char_t* host_path,
+    const pal::char_t* dotnet_root,
+    const pal::char_t* app_path,
+    void **delegate);
+using hostfxr_main_fn = int(*)(const int argc, const pal::char_t* argv[]);
+using hostfxr_main_startupinfo_fn = int(*)(
+    const int argc,
+    const pal::char_t* argv[],
+    const pal::char_t* host_path,
+    const pal::char_t* dotnet_root,
+    const pal::char_t* app_path);
+using hostfxr_error_writer_fn = void(*)(const pal::char_t* message);
+using hostfxr_set_error_writer_fn = hostfxr_error_writer_fn(*)(hostfxr_error_writer_fn error_writer);
```

Comment:
I would prefer if we would separate the function pointers into their own header. They don't really belong to the resolver.
Maybe `hostfxr.h`? (Might even ship it for people to easily include one day)